addons/account/models/account_account.py

```diff
@@ -140,6 +140,41 @@ def _field_to_sql(self, alias: str, fname: str, query: (Query | None) = None, fl
             return SQL("split_part(account_account.account_type, '_', 1)", to_flush=self._fields['account_type'])
         if fname == 'code':
             return self.with_company(self.env.company.root_id).sudo()._field_to_sql(alias, 'code_store', query, flush)
+        if fname == 'placeholder_code':
+            query.add_join(
+                'JOIN',
+                'account_first_company',
+                SQL(
+                    """(
+                        SELECT DISTINCT ON (rel.account_account_id)
+                               rel.account_account_id AS account_id,
+                               rel.res_company_id AS company_id,
+                               SPLIT_PART(res_company.parent_path, '/', 1) AS root_company_id,
+                               res_company.name AS company_name
+                          FROM account_account_res_company_rel rel
+                          JOIN res_company
+                            ON res_company.id = rel.res_company_id
+                         WHERE rel.res_company_id IN %(authorized_company_ids)s
+                      ORDER BY rel.account_account_id, company_id
+                    )""",
+                    authorized_company_ids=self.env.user._get_company_ids(),
+                ),
+                SQL('account_first_company.account_id = %(account_id)s', account_id=SQL.identifier(alias, 'id')),
+            )
+
+            # Can't have spaces because of how stupidly the `Model._read_group_orderby` method is written
+            # see https://github.com/odoo/odoo/blob/2a3466e8f86bc08594391658e08ba3416fb8307b/odoo/models.py#L2222
+            return SQL(
+                "COALESCE("
+                "%(code_store)s->>%(active_company_root_id)s,"
+                "%(code_store)s->>%(account_first_company_root_id)s||'('||%(account_first_company_name)s||')'"
+                ")",
+                code_store=SQL.identifier(alias, 'code_store'),
+                active_company_root_id=self.env.company.root_id.id,
```

Comment:
A small typo here
it should be `active_company_root_id=str(self.env.company.root_id.id),`
since `SELECT '{"1": "100", "11": "200"}'::jsonb->>1;` is `NULL`

Reply:
To be fixed by https://github.com/odoo/odoo/pull/183761